src/time-picker/index.tsx

```diff
@@ -12,6 +12,19 @@ import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
 import { Keys } from '../common/util';
 import bundle from './nls/TimePicker';
 import i18n from '@dojo/framework/core/middleware/i18n';
+import { createResource, DataTemplate } from '@dojo/framework/core/resource';
+
+function createMemoryTemplate<S = void>(): DataTemplate<S> {
+	return {
+		read: ({ query }, put, get) => {
+			let data: any[] = get();
+			put(0, data);
```

Comment:
Why do we need to `get` and then immediately `put` the same data?

Reply:
The `read` function gets passed  both a getter and a setter function. In the case of a memory template, the first request may only call `read` for the first few results, but we wish to get all of the data and set it back again against the query that was passed. This simple template is not the best example of this functionality really but it _does_ begin to make more sense when filters etc are concerned. Basically, for a select etc, it ensure that all the items are available and have been sideloaded into the widget meaning items can be navigated to via `a-z` press etc.